src/index.adoc

```diff
@@ -4,14 +4,542 @@ include::header.adoc[]
+bad.
+
+=== link:https://www.youtube.com/watch?v=JjDsP5n2kSM&t=2169s[Jonathan Blow: How
+to program independent games^] You don't succeed by using fancy data structure or
+writing an algorithm that's a bit faster but buggy and fragile. Independent game
+developer needs to be "extremely effective at getting things done".
+
+=== Rich Hickey on immutable values versus references
+
+- link:https://www.infoq.com/presentations/Value-Values[The Value of Values^]
+- link:https://www.infoq.com/presentations/Are-We-There-Yet-Rich-Hickey[Are we
+  there yet^]
+
+This is Rich (again!) promoting immutable values and immutable data structures
+over references that change over time in a mutable manner. Also, Rich talks
+about the Epochal time model, where the whole (!) state of the program is
+represented by a single mutable reference that takes a sequence of immutable
+values, modeling a time-evolution of a state. This model is used with Redux
+framework.
+
+=== link:https://www.youtube.com/watch?v=FihU5JxmnBg[Stuart Halloway: Debugging
+with the Scientific Method^]
+
+Be less chaotic and more organized when debugging. State explicitly what's your
+hypothesis, test it, think. Write things down. Test one thing at a time. A good
+process is more important than tools; scientific approach with console.log is
+more powerful than ad-hoc trying random stuff with a strong debugger.
 
 == Tools
-include::tools/index.adoc[]
+
+=== Operating System
+Use either Mac or Linux for development. Don't use Windows, don't even think
+about it! There is always something working differently on Windows than on the
+other platforms. This costs us huge amounts of time debugging problems instead
+of developing valuable features.
+
+If you are new to Linux, install link:http://www.ubuntu.com[Ubuntu^]. Make sure to read
+link:https://help.ubuntu.com/stable/ubuntu-help[the introduction to Ubuntu Desktop^], especially
+link:https://help.ubuntu.com/stable/ubuntu-help/addremove.html[how the software installation work in Linux^].
+
+// ==== Linux
+
+==== macOS
+
+Get a _i_**Mac**_Book_ _Pro_. macOS can run on non-Apple computer, it's called
+Hackintosh, but things can work differently there and it's against EULA. 😞
+
+macOS has integrated App Store from which you can download some useful
+applications like XCode, Slack, and iWork (Pages, Numbers and Keynote) apps that
+came with your new Mac. You will use App Store to update the system, too.
+
+macOS doesn't come with package manager as is `apt` in the Ubuntu. There are two
+options link:https://brew.sh[Homebrew^] and
+link:https://www.macports.org[MacPorts^]. The most used in our company is
+Homebrew.
+
+=== Terminal
+
+Learn link:http://linuxcommand.org/learning_the_shell.php[the basics of UNIX shell^].
+
+Replace your default shell with much better one called _zsh_. Install
+link:https://ohmyz.sh/[oh-my-zsh^], where you can select some nice visual theme for your shell.
```

Comment:
Imho for beginners, bash should be the standard. Most guides and tools work really well with that. With zsh even on mac, a lot of tools have hiccups, bad installations, bad env setup. On linux it made me switch away from it. Last time we ran into that was with `volta.sh` . You can install powerline for bash as well.
https://github.com/powerline/powerline
https://medium.com/earlybyte/powerline-for-bash-6d3dd004f6fc